metaviews/src/iosMain/kotlin/com/mirego/trikot/metaviews/lifecycle/ApplicationStatePublisher.kt

```diff
@@ -0,0 +1,51 @@
+package com.mirego.trikot.metaviews.lifecycle
+
+import com.mirego.trikot.streams.reactive.BehaviorSubjectImpl
+import org.reactivestreams.Publisher
+import platform.Foundation.NSNotificationCenter
+import platform.UIKit.UIApplication
+import platform.UIKit.UIApplicationDidEnterBackgroundNotification
+import platform.UIKit.UIApplicationState
+import platform.UIKit.UIApplicationWillEnterForegroundNotification
+import platform.darwin.sel_registerName
+
+actual class ApplicationStatePublisher : BehaviorSubjectImpl<ApplicationState>(),
+    Publisher<ApplicationState> {
+    init {
+        value = when (UIApplication.sharedApplication.applicationState) {
```

Comment:
Juste une note que `UIApplication.sharedApplication.applicationState` doit etre lu seulement sur le main thread.

Donc le init de cet ApplicationStatePublisher doit se faire seulement sur le main thread.

Ca fonctionne pareil sur un background thread, mais c'est un comportement indetermine et Xcode fait des warnings.

Reply:
Est-ce qu'on pourrait se créer une petite classe expect/actual qui ferait le chech de "main thread", et l'utiliser ici pour être sur qu'on est dans le bon cas?

Reply:
No need to use expect/actual here since we are already on the ios target here.

I addressed your concern in 
https://github.com/mirego/trikot.metaviews/pull/40/commits/b857784e6e6f4fc79df41d40d91bf0bef4a8583d

It does now dispatch directly when on main queue and queue to main when in background.

I've tested creating the Publisher from background and it works as expected.

I would rather have used a UIDispatcher (we have one in our project but haven't had the time to exact it to trikot.streams just yet). That might be done later on.